adagrams/game.py

```diff
@@ -1,11 +1,107 @@
+import random
+# ?What is your time complexity?
 def draw_letters():
-    pass
+
+    ada_gram = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+    
+    letter_pool = []
+    hand_as_list = []
+    # should this be skipped and the data structure begin in list format?
+    for k,v in ada_gram.items():
+        for i in range(v):
+            letter_pool.append(k)
+
+    while len(hand_as_list) < 10:
+        random_letter = random.choice(letter_pool)
+        hand_as_list.append(random_letter)
+        letter_pool.remove(random_letter)
+
+    return hand_as_list
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letter_list = []
 
+    for char in letter_bank:
+        letter_list.append(char)
+    
+    for letter in word:
+        upper_letter = letter.upper()
+        if upper_letter in letter_list:
+            letter_list.remove(upper_letter)
+        else:
+            return False
+    return True
+
+# ?What is your time complexity#
 def score_word(word):
-    pass
 
+    score_chart = {
+        1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+        2: ["D", "G"],
+        3: ["B", "C", "M", "P"],
+        4: ["F", "H", "V", "W", "Y"],
+        5: ["K"],
+        8: ["J", "X"],
+        10: ["Q", "Z"]
+    }
+
+    score =  0
+    
+    for k,v in score_chart.items():
+        for letter in word:
+            if letter.upper() in v:
+                score += k
```

Comment:
The code here can be more efficient if the scores were stored the same way the letter values are in the `ada_gram` dictionary. That way you can look up each letter as the key to the dictionary rather than needing to loop through all of the values.